wandb/sdk/wandb_run.py

```diff
@@ -3072,6 +3075,93 @@ def _prepare_artifact(
         artifact.finalize()
         return artifact, _resolve_aliases(aliases)
 
+    def log_model(
+        self,
+        local_path: StrPath,
+        model_name: Optional[str] = None,
+        aliases: Optional[List[str]] = None,
+    ) -> None:
+        """Declare a model artifact as an output of a run.
+
+        Arguments:
+            local_path: (str) A path to the contents of this model,
+                can be in the following forms:
+                    - `/local/directory`
+                    - `/local/directory/file.txt`
+                    - `s3://bucket/path`
+            model_name: (str, optional) An artifact name. May be prefixed with entity/project.
+                Valid names can be in the following forms:
+                    - name:version
+                    - name:alias
+                    - digest.
+                This will default to the basename of the path prepended with the current
+                run id  if not specified.
+            aliases: (list, optional) Aliases to apply to this artifact,
+                    defaults to `["latest"]`
+
+        Returns:
+            None
+        """
+        self._log_artifact(
+            artifact_or_path=local_path, name=model_name, type="model", aliases=aliases
+        )
+        return None
+
+    @_run_decorator._attach
+    def use_model(self, model_name: str) -> StrPath:
+        """Download a logged model artifact.
+
+        Arguments:
+            model_name: (str) A model artifact name.
+                May be prefixed with entity/project/. Valid names
+                can be in the following forms:
+                    - name:version
+                    - name:alias
+                    - digest.
+        """
+        r = self._run_obj
+        assert r is not None
+        artifact = self.use_artifact(artifact_or_name=model_name)
+        assert artifact.type.lower() == "model"
+        local_path = artifact.download()
+        return local_path
```

Comment:
Nit: `use_artifact()` returns an `Artifact`, but `use_model()` returns a directory path. This is confusing.